src/Workspaces/CSharp/Portable/Classification/SyntaxClassification/DiscardDesignationSyntaxClassifier.cs

```diff
@@ -0,0 +1,28 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using Microsoft.CodeAnalysis.Classification;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.PooledObjects;
+
+namespace Microsoft.CodeAnalysis.CSharp.Classification.Classifiers
+{
+    internal class DiscardDesignationSyntaxClassifier : AbstractSyntaxClassifier
+    {
+        public override void AddClassifications(
+            SyntaxNode syntax,
+            SemanticModel semanticModel,
+            ArrayBuilder<ClassifiedSpan> result,
+            CancellationToken cancellationToken)
+        {
+            if (syntax is DiscardDesignationSyntax discardDesignation)
```

Comment:
isn't this redundant?  i.e. shoudln't this just be a straight cast given that you only register for DiscardDesignationSyntax?